src/Apps/gAtmoEvGen.cxx

```diff
@@ -350,7 +358,10 @@ int main(int argc, char** argv)
   }
 
   // get flux driver
-  GFluxI * flux_driver = GetFlux();
+  atmo_flux_driver = GetFlux();
+
+  // Cast to GFluxI, the generic flux driver interface
+  GFluxI *flux_driver = dynamic_cast<GFluxI *>(atmo_flux_driver);
```

Comment:
pedantically I'm not sure this is ever necessary to down(?) cast to a base class unless the derived class hides some base class method.   You should be able to just call those base class methods on the derived class or pass derived object pointer to methods that expect the base class.   But that was an issue with the original code, so not a required change here.

Reply:
Good point, I will update this and push a new set of changes.